File: app/src/main/java/com/shevelev/comics_viewer/activities/main/bookshelf/ComicsClickListener.java
package com.shevelev.comics_viewer.activities.main.bookshelf;

import android.view.View;

import com.shevelev.comics_viewer.common.func_interfaces.IActionOneArgs;

public class ComicsClickListener implements View.OnClickListener
{
    private long dbComicsId;

    private IActionOneArgs<Long> onComicsChoosen;

    public ComicsClickListener(long dbComicsId, IActionOneArgs<Long> onComicsChoosen)
    {
        this.dbComicsId = dbComicsId;
        this.onComicsChoosen = onComicsChoosen;
    }

    @Override
    public void onClick(View v)
    {
        onComicsChoosen.process(dbComicsId);
    }
}

File: app/src/main/java/com/shevelev/comics_viewer/activities/main/bookshelf/ComicsClickListener.kt
package com.shevelev.comics_viewer.activities.main.bookshelf

import android.view.View

class ComicsClickListener(private val dbComicsId: Long, private val onComicsChoosen: (Long) -> Unit) : View.OnClickListener {
    override fun onClick(v: View) {
        onComicsChoosen(dbComicsId)
    }
}